homa_incoming.c

```diff
@@ -681,14 +681,31 @@ void homa_resend_pkt(struct sk_buff *skb, struct homa_rpc *rpc,
 			rpc->id, ntohl(h->offset), ntohl(h->length),
 			h->priority);
 
-	if (!homa_is_client(rpc->id)) {
+	if (!homa_is_client(rpc->id) && rpc->state != RPC_OUTGOING) {
 		/* We are the server for this RPC. */
-		if (rpc->state != RPC_OUTGOING) {
+		if ((rpc->msgin.length - rpc->msgin.bytes_remaining) >= rpc->msgin.granted) {
```

Comment:
Follow Linux kernel coding style requirements and maintain consistency with the rest of the Homa code: no lines longer than 80 characters.